spotinst/resource_spotinst_ocean_spark.go

```diff
@@ -195,27 +198,81 @@ func resourceSpotinstSparkClusterDelete(ctx context.Context, resourceData *schem
 		return diag.FromErr(err)
 	}
 
-	log.Printf("===> Cluster deleted successfully: %s <===", resourceData.Id())
+	log.Printf("===> cluster deleted successfully: %s <===", resourceData.Id())
 	resourceData.SetId("")
 
 	return nil
 }
 
 func deleteSparkCluster(ctx context.Context, resourceData *schema.ResourceData, meta interface{}) error {
 	clusterID := resourceData.Id()
+	forceDelete := false
+	if os.Getenv("TF_ACC") == "1" {
+		log.Printf("===> force delete set to true for integration tests")
+		forceDelete = true
+	}
+
 	input := &spark.DeleteClusterInput{
-		ClusterID: spotinst.String(clusterID),
+		ClusterID:   spotinst.String(clusterID),
+		ForceDelete: spotinst.Bool(forceDelete),
 	}
 
 	if json, err := commons.ToJson(input); err != nil {
 		return err
 	} else {
-		log.Printf("===> Cluster delete configuration: %s", json)
+		log.Printf("===> cluster delete configuration: %s", json)
 	}
 
 	if _, err := meta.(*Client).ocean.Spark().DeleteCluster(ctx, input); err != nil {
 		return fmt.Errorf("[ERROR] onDelete() -> Failed to delete cluster: %s", err)
 	}
 
+	if forceDelete {
+		log.Printf("===> cluster has been force deleted: %s <===", resourceData.Id())
+		return nil
+	}
+
+	if err := waitUntilClusterDeleted(ctx, resourceData, meta); err != nil {
+		return fmt.Errorf("[ERROR] onDelete() -> failed to complete delete cluster: %s", err)
+	}
+
 	return nil
 }
+
+func waitUntilClusterDeleted(ctx context.Context, resourceData *schema.ResourceData, meta interface{}) error {
+	timeout := time.After(deleteTimeout)
+	for {
+		select {
+		case <-timeout:
+			return fmt.Errorf("[ERROR] waitUntilClusterDeleted() -> Cluster %s was not deleted within timeout", resourceData.Id())
+		default:
+			isDeleted, err := isClusterDeleted(ctx, resourceData, meta)
+			if err != nil {
+				return fmt.Errorf("[ERROR] waitUntilClusterDeleted() -> Failed to delete cluster: %s %w", resourceData.Id(), err)
+			}
+
+			if isDeleted {
+				log.Printf("===> Cluster %s was deleted within timeout :  <===", resourceData.Id())
```

Comment:
we can remove this log string I think

we have the `log.Printf("===> Cluster deleted successfully: %s <===", resourceData.Id())` log string